### PR TITLE
Better Error Pages

### PR DESCRIPTION
**Pull request by EnBi81**
Creating better error pages. I created a hook, so it is easy to redirect the user to the error page. Also added an errorCode parameter to the useRequest, so that the website can redirect to user to the appropriate error page.

### CODE REVIEW

**Comment by EnBi81** [comment]
So this branch has a lot of commits, because I merged the terrarium branch into this one, because I needed something from that branch. But basically check changes from the commit message [adding statusCode and errorCode to useRequest]

**Comment by EnBi81** [comment]
Actually we could use this pull request to merge the terrarium branch into the master, and if somebody wants to take the animals task because they feel like they want to contribute more to the git repository, then feel free to take it

File: libs/measurement-history/src/lib/components/PieChartBoundaries.tsx
Comment (Alzimax) [inline libs/measurement-history/src/lib/components/PieChartBoundaries.tsx:1]: Maybe you could write just data?.length > 0 for the check

File: libs/measurement-history/src/lib/measurement-history.tsx
Comment (Alzimax) [inline libs/measurement-history/src/lib/measurement-history.tsx:1]: Maybe also write measurmentRange?.length > 0 for the check

**Review by Alzimax — APPROVED** [review]

**Review by apeTrackunit — APPROVED** [review]

**Pull request merged**